src/app/modules/users/users-list/users-list.component.ts

```diff
@@ -1,100 +1,297 @@
+      // Time in milliseconds between key events
+      , debounceTime(1000)
+      // If previous query is diffent from current
+      , distinctUntilChanged()
+      // subscription for response
+    ).subscribe((text: string) => {
+      this.getUserList();
+    });
+    this.getUserOrginasations();
+    this.paginator.page.subscribe((page: PageEvent) => {
+      this.queryParams.page = page.pageIndex + 1;
+      this.queryParams.size = page.pageSize;
+      this.getUserList();
+    });
+    this.createForm();
+  }
 
+  /**
+* To get Userslist
+*/
+  getUserList() {
+    this.usersService.getUsers(this.queryParams, this.orgnsationId, this.searchInput.nativeElement.value).subscribe(data => {
+      this.options = data['result'];
+      this.refreshDatasource(data['result'].data);
+      this.dataSource = new MatTableDataSource(data['result'].data);
+      // this.dataSource.paginator = this.paginator;
+      this.dataSource.sort = this.sort;
+      this.recordcount = data['result'].count;
+      this.cdr.detectChanges();
+      this.listing = true;
+      console.log('lissssssssssss', this.options);
+    }, error => {
+
+    });
+  }
+  refreshDatasource(data) {
+    this.dataSource = data;
+    // this.cdr.detectChanges();
+  }
+
+  selectorganisation() {
+    this.orgnsationId = this.selected;
+    this.getUserList();
+  }
+
+  onKey(value) {
+    this.organisations = this.search(value);
+    this.cdr.detectChanges();
+  }
+
+  // Filter the states list and send back to populate the selectedStates**
+  search(value) {
+    let filter = value.toLowerCase();
+    return this.organisationList.filter(option => option.label.toLowerCase().startsWith(filter));
+  }
+  /** Whether the number of selected elements matches the total number of rows. */
+  isAllSelected() {
+    const numSelected = this.selection.selected.length;
+    // console.log('isAllSelected', numSelected);
+    if (this.dataSource) {
+      const numRows = this.dataSource.data.length;
+      return numSelected === numRows;
+    }
+
+  }
+  /** Selects all rows if they are not all selected; otherwise clear selection. */
+  masterToggle() {
+    this.isAllSelected() ?
+      this.selection.clear() :
+      this.dataSource.data.forEach(row => this.selection.select(row));
+    // console.log('masterToggle', this.dataSource);
+  }
+
+  /** The label for the checkbox on the passed row */
+  checkboxLabel(row): string {
+    // console.log('checkboxLabel', row);
+    if (!row) {
+      return `${this.isAllSelected() ? 'select' : 'deselect'} all`;
+    }
+    return `${this.selection.isSelected(row) ? 'deselect' : 'select'} row ${row.length + 1}`;
+  }
+
+  /**
+   * To get the form from the backend
+   */
+  getUserOrginasations() {
+    this.usersService.getOrganisations().subscribe(data => {
+      this.options = data['result'];
+      console.log('getUserOrginasations', this.options);
+      this.organisations = data['result'];
+      this.organisationList = data['result'];
```

Comment:
why 2 variables with the same value

Reply:
for search in dropdown issue i kept, i will remove once it is fixed